test/e2e/generate_incident.go

```diff
@@ -0,0 +1,201 @@
+package osde2etests
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+type PagerDutyClient interface {
+	CreateSilentRequest(alertName, clusterID string) (string, error)
+	GetIncidentID(dedupKey string) (string, error)
+	ResolveIncident(incidentID string) error
+}
+
+type client struct {
+	eventsURL     string
+	apiURL        string
+	routingKey    string
+	authToken     string
+	alertMappings map[string]string
+	httpClient    *http.Client
+}
+
+func NewClient(routingKey, authToken string) PagerDutyClient {
+	return &client{
+		eventsURL:  "https://events.pagerduty.com/v2/enqueue",
+		apiURL:     "https://api.pagerduty.com/incidents",
+		routingKey: routingKey,
+		authToken:  authToken,
+		alertMappings: map[string]string{
+			"ClusterHasGoneMissing":                         "cadtest has gone missing",
+			"ClusterProvisioningDelay":                      "ClusterProvisioningDelay -",
+			"ClusterMonitoringErrorBudgetBurnSRE":           "ClusterMonitoringErrorBudgetBurnSRE Critical (1)",
+			"InsightsOperatorDown":                          "InsightsOperatorDown",
+			"MachineHealthCheckUnterminatedShortCircuitSRE": "MachineHealthCheckUnterminatedShortCircuitSRE CRITICAL (1)",
+			"ApiErrorBudgetBurn":                            "api-ErrorBudgetBurn k8sgpt test CRITICAL (1)",
+		},
```

Comment:
Let's avoid having to pass raw strings. 

Suggestion:
```
const (
	ClusterHasGoneMissing                = "ClusterHasGoneMissing"
	ClusterProvisioningDelay             = "ClusterProvisioningDelay"
	ClusterMonitoringErrorBudgetBurnSRE  = "ClusterMonitoringErrorBudgetBurnSRE"
	// ...
)

func AlertTitle(alertName string) string {
	switch alertName {
	case ClusterHasGoneMissing:
		return "cadtest has gone missing"
	case ClusterProvisioningDelay:
		return "ClusterProvisioningDelay -"
	// ...
	default:
		return "Unknown Alert"
	}
}
```

The client we are creating here does not need to contain this mapping internally.